pkg/service/common/machine.go

```diff
@@ -80,6 +81,16 @@ func init() {
 	providerInstanceType[constants.AzureLabel] = azure()
 	providerInstanceType[constants.AwsLabel] = aws()
 	providerInstanceType[constants.GcpLabel] = gcp()
+
+	gpus = map[string]bool{
```

Comment:
`map[string]struct{}` would do and has a lesser footprint.

Reply:
sorry, I din get. Can you elaborate ? 
bool was chosen specifically to help the lookup without using `_, ok` pattern.

Reply:
Bool still leads to memory allocation, usually, if we're looking for something like a set then map[T]struct{} is a better way.

Reply:
it would make lookup ugly, i want to avoid that. not the allocations. Function using this will simpy returns
```
func IsGpu(m string){
  return gpus[m]
}
```
instead of
```
func IsGpu(m string){
  if _, ok := gpus[m]; ok {
    return ok
  }
  return false
}
```

Reply:
It can just be
`
func IsGpu(m string){
  _, ok := gpus[m]
  return ok
}
`

Reply:
true, the above looks much cleaner without `_`. If it was a long list then yeah, its worth the concern for allocation. 
Wdyt ?

Reply:
The said struct is good for dynamic or long list of set. Verbosity of that for this static list isnt really  necessary.

Reply:
Anyhow, will update it to struct{}